### ISSUE TITLE
Zero denominator: numpy returns NaN while numba/Metal/CUDA return 0.0 — the backend choice changes the scientific result

### ISSUE THREAD

**Issue by Ramdam17** [ppsp-team/HyPyP#297]
The NumPy reference implementations do **not** guard against a zero denominator, while every numba/Metal/CUDA kernel does (`if denom > 0 ... else 0.0`). With a flat channel — a dead channel, or one zeroed/interpolated during preprocessing — the two paths disagree.

Measured on a `(1, 1, 6, 300)` complex signal with channel 3 set to `0+0j`:

```
  coh      numpy=[nan nan nan nan]  |  numba=[0. 0. 0. 0.]
  imcoh    numpy=[nan nan nan nan]  |  numba=[0. 0. 0. 0.]
  envcorr  numpy=[nan nan nan nan]  |  numba=[0. 0. 0. 0.]
  powcorr  numpy=[nan nan nan nan]  |  numba=[0. 0. 0. 0.]
  ccorr    numpy=[nan nan nan nan]  |  numba=[0. 0. 0. 0.]
```

**Why this is more than an edge case.** `compute_sync` finishes with `np.nanmean(con, axis=1)` over epochs (`hypyp/analyses.py:558`). On the NumPy path the contaminated epoch is *excluded* from the average; on the numba/GPU path a `0.0` is *included* and drags the mean down. The two backends therefore do not differ by a numerical epsilon — they differ by a systematic negative bias whose magnitude scales with the number of affected epochs. Switching `optimization=` changes the reported connectivity.

Guarded (return 0.0):
- `hypyp/sync/coh.py:168`, `hypyp/sync/imaginary_coh.py:173`, `hypyp/sync/envelope_corr.py:178`, `hypyp/sync/pow_corr.py` (numba kernels)
- `hypyp/sync/ccorr.py:235`
- all CUDA kernels, e.g. `hypyp/sync/kernels/cuda_amplitude.py:52`, `:95`, `:148`, `:201`
- Metal: `hypyp/sync/kernels/metal_accorr.py:89`

Unguarded (produce NaN/inf):
- `hypyp/sync/coh.py:89-90`, `hypyp/sync/imaginary_coh.py:92-93`, `hypyp/sync/envelope_corr.py:93-94`, `hypyp/sync/pow_corr.py:95-96`, `hypyp/sync/ccorr.py:99-101`

Related but separate inconsistency: `Coh._compute_numpy` uses `np.nansum` for the power term (`coh.py:89`) but an `einsum` that propagates NaN for the numerator, while the numba kernel uses a plain sum and `_compute_torch` uses `torch.sum` (`coh.py:126`). The `nansum` is only half-applied.

**Decision needed before fixing** — this is methodological, not just an implementation detail. Should a non-measurable pair be `NaN` (honest, properly excluded by the downstream `nanmean`, but propagates silently) or `0.0` (robust, but asserts an absence of synchrony that was never measured, and is included in the mean)? Whichever is chosen, it must be applied identically across all five backends, and a `warnings.warn` naming the offending channels would help either way.
